main.c

```diff
@@ -1,78 +1,35 @@
-};
-int numberOfMajorColors =
-    sizeof(MajorColorNames) / sizeof(MajorColorNames[0]);
-const char* MinorColorNames[] = {
-    "Blue", "Orange", "Green", "Brown", "Slate"
-};
-const int MAX_COLORPAIR_NAME_CHARS = 16;
-int numberOfMinorColors =
-    sizeof(MinorColorNames) / sizeof(MinorColorNames[0]);
+/*
+ * main.c
+ *
+ *  Created on: Feb 22, 2021
+ *      Author: PUL1KOR
+ */
 
-typedef struct {
-    enum MajorColor majorColor;
-    enum MinorColor minorColor;
-} ColorPair;
-
-void ColorPairToString(const ColorPair* colorPair, char* buffer) {
-    sprintf(buffer, "%s %s",
-        MajorColorNames[colorPair->majorColor],
-        MinorColorNames[colorPair->minorColor]);
-}
-
-ColorPair GetColorFromPairNumber(int pairNumber) {
-    ColorPair colorPair;
-    int zeroBasedPairNumber = pairNumber - 1;
-    colorPair.majorColor = 
-        (enum MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
-    colorPair.minorColor =
-        (enum MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
-    return colorPair;
-}
-
-int GetPairNumberFromColor(const ColorPair* colorPair) {
-    return colorPair->majorColor * numberOfMinorColors +
-            colorPair->minorColor + 1;
-}
 
-void testNumberToPair(int pairNumber,
-    enum MajorColor expectedMajor,
-    enum MinorColor expectedMinor)
-{
-    ColorPair colorPair = GetColorFromPairNumber(pairNumber);
-    char colorPairNames[MAX_COLORPAIR_NAME_CHARS];
-    ColorPairToString(&colorPair, colorPairNames);
-    printf("Got pair %s\n", colorPairNames);
-    assert(colorPair.majorColor == expectedMajor);
-    assert(colorPair.minorColor == expectedMinor);
-}
-
-void testPairToNumber(
-    enum MajorColor major,
-    enum MinorColor minor,
-    int expectedPairNumber)
+#include <stdio.h>
+#include "CablePairTester_CommonCfg.h"
+#include "CablePairNumber_Tetser.h"
+#include "CableColourPair_Tetser.h"
+
+/********************
+ *
+ * Des: Main function. This will call the Colour pair testing functions.
+ * Input : None
+ * Output: None
+ *
+ */
+
+int main()
 {
-    ColorPair colorPair;
-    colorPair.majorColor = major;
-    colorPair.minorColor = minor;
-    int pairNumber = GetPairNumberFromColor(&colorPair);
-    printf("Got pair number %d\n", pairNumber);
-    assert(pairNumber == expectedPairNumber);
-}
-
-int main() {
-    testNumberToPair(4, WHITE, BROWN);
-    testNumberToPair(5, WHITE, SLATE);
-
-    testPairToNumber(BLACK, ORANGE, 12);
-    testPairToNumber(VIOLET, SLATE, 25);
-
+	/*********
+	 * available major colours :  WHITE, RED, BLACK, YELLOW, VIOLET
+	 * available minor colours :  BLUE, ORANGE, GREEN, BROWN, SLATE
+	 * other colours are invalid. Passing will return error.
+	 * "https://en.wikipedia.org/wiki/25-pair_color_code" fore more details
+	 */
+    Fc_testNumberToPair(4, WHITE, BROWN);
+    Fc_testNumberToPair(5, WHITE, SLATE);
+    Fc_testPairToNumber(BLACK, ORANGE, 12);
+    Fc_testPairToNumber(VIOLET, SLATE, 25);
     return 0;
```

Comment:
The functionality in main.c is to test. Should it be called 'test...'?
As I see, you have not yet added the functionality to print the reference manual. Where would you add that?